default.nix

```diff
@@ -13,6 +13,120 @@ let
 
   cargoToml = with builtins; (fromTOML (readFile ./Cargo.toml));
 
+  # A new kind of option type that calls lib.getExe on derivations
+  exeType = lib.mkOptionType {
+    name = "exe";
+    description = "Path to executable";
+    check = (x: lib.isString x || lib.isPath x || lib.isDerivation x);
+    merge = loc: defs:
+      let res = lib.mergeOneOption loc defs; in
+      if lib.isString res || lib.isPath res then
```

Comment:
When passing a package, this is failing with:

```
error: attribute 'isPath' missing

       at /nix/store/n7v25mkag2k3g170ljfkppg5aj07hqv5-source/default.nix:20:35:

           19|     description = "Path to executable";
           20|     check = (x: lib.isString x || lib.isPath x || lib.isDerivation x);
             |                                   ^
           21|     merge = loc: defs:
```

Should it be `builtins.isPath` instead? However, `lib.isPath` should work in theory, going by [what's in nixpkgs](https://github.com/NixOS/nixpkgs/search?q=isPath&type=code).